pkg/git/libgit2/transport.go

```diff
@@ -126,20 +128,24 @@ func (s *PublicKeyAuth) Method(secret corev1.Secret) (*git.Auth, error) {
 		user = git.DefaultPublicKeyAuthUser
 	}
 
-	credCallback := func(url string, username_from_url string, allowed_types git2go.CredType) (*git2go.Cred, error) {
+	credCallback := func(url string, usernameFromURL string, allowedTypes git2go.CredType) (*git2go.Cred, error) {
 		cred, err := git2go.NewCredSshKeyFromMemory(user, "", string(identity), "")
 		if err != nil {
 			return nil, err
 		}
 		return cred, nil
 	}
 	certCallback := func(cert *git2go.Certificate, valid bool, hostname string) git2go.ErrorCode {
+		host, _, err := net.SplitHostPort(s.host)
+		if err != nil || host != hostname {
+			return git2go.ErrUser
+		}
 		for _, k := range kk {
-			if k.matches(hostname, cert.Hostkey.HashSHA1[:]) {
+			if k.matches(s.host, cert.Hostkey.HashSHA1[:]) {
 				return git2go.ErrOk
 			}
 		}
-		return git2go.ErrGeneric
+		return git2go.ErrUser
```

Comment:
Would ErrAuth or ErrCertificate be a better choice?

Reply:
I was wondering that too while looking through the error codes, but I can not find what type of certificate they mean...

Reply:
```
// ErrorCodeCertificate represents that the server certificate is invalid.
```

Oh doh...